src/containers/PerfherderGraphContainer/index.jsx

```diff
@@ -22,19 +22,26 @@ const styles = () => ({
 class PerfherderGraphContainer extends Component {
   state = {
     data: null,
+    isLoading: false,
   };
 
   async componentDidMount() {
     this.fetchSetData(this.props);
   }
 
   async fetchSetData({ series }) {
-    this.setState(await getPerferherderData(series));
+    try {
+      this.setState({ isLoading: true });
+      this.setState(await getPerferherderData(series));
+      this.setState({ isLoading: false });
+    } catch (error) {
+      this.setState({ isLoading: false });
```

Comment:
Shouldn't we do something with this error - store it in state and then use it to show `ErrorPanel`? 

`this.setState({ isLoading: false });` can be moved outside the try/catch block since it will be called either way.

Reply:
Technically it may behave asynchronously, but that's not why I was mentioning it :) Calling setState out of an event handle used to cause a new render for each time it's called but that is apparently not the case as of v16+: https://itnext.io/react-setstate-usage-and-gotchas-ac10b4e03d60

Reply:
I have pushed the updated code. Please let me know if it looks good now.

Reply:
The changes look good Aimen, but I'm still wondering why we aren't catching the errors and showing an ErrorPanel here and in the BugzillaGraph component?

Reply:
@sarah-clements the error catching will be caught in my next PR: https://github.com/mozilla-frontend-infra/firefox-health-dashboard/pull/330